docs/reference/python/conventions.md

```diff
@@ -18,6 +18,102 @@ pep8 style guide.
     Our CI automatically checks that your code follows the pep8 standard. If it does not, your pull requests will
     be blocked from being merged until those issues are resolved!
 
+### Type hinting
+
+Even though Python is a dynamically typed language, newer versions support type hinting. Type hinting catches
+errors, documents code, improves IDEs and linters, and helps build and maintain a clean software architecture.[^1]
+Expanding on how it catches errors, a static type checker such as [`mypy`](https://mypy.readthedocs.io/en/stable/index.html){target=_blank}
+can be used.
+
+There are [pros and cons](https://realpython.com/lessons/pros-and-cons-type-hints/){target=_blank} to using type
+hinting, make sure that you are aware of them before deciding to utilize them in your codebase.
+
+There is some syntax to get familiar in order to use type checking. We recommend the following resources:
+
+- [mypy Typing Cheatsheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html){target=_blank}
+- [PEP 483: The Theory of Type Hints (A Simplified Guide)](https://peps.python.org/pep-0483/){target=_blank}
+- [PEP 484: Type Hints (Fully Comprehensive Guide)](https://peps.python.org/pep-0484/){target=_blank}
+
+Below are a few examples of using type hinting:
+
+??? example "Return the sum of a sequence"
```

Comment:
To make it easier to skim read, consider leaving 1 or 2 examples open by default. Can do the same for documentation examples.